makefiles/suit.v4.inc.mk

```
@@ -0,0 +1,103 @@
+#
+SUIT_COAP_BASEPATH ?= fw/$(BOARD)
+SUIT_COAP_SERVER ?= localhost
+SUIT_COAP_ROOT ?= coap://$(SUIT_COAP_SERVER)/$(SUIT_COAP_BASEPATH)
+SUIT_COAP_FSROOT ?= $(RIOTBASE)/coaproot
+
+#
+SUIT_MANIFEST ?= $(BINDIR_APP)-riot.suitv4.$(APP_VER).bin
+SUIT_MANIFEST_LATEST ?= $(BINDIR_APP)-riot.suitv4.latest.bin
+SUIT_MANIFEST_SIGNED ?= $(BINDIR_APP)-riot.suitv4_signed.$(APP_VER).bin
+SUIT_MANIFEST_SIGNED_LATEST ?= $(BINDIR_APP)-riot.suitv4_signed.latest.bin
+
+SUIT_NOTIFY_VERSION ?= latest
+SUIT_NOTIFY_MANIFEST ?= $(BINDIR_APP)-riot.suitv4_signed.$(SUIT_NOTIFY_VERSION).bin
+
+# Long manifest names require more buffer space when parsing
+export CFLAGS += -DSOCK_URLPATH_MAXLEN=128
+
+SUIT_VENDOR ?= "riot-os.org"
+SUIT_SEQNR ?= $(APP_VER)
+SUIT_CLASS ?= $(BOARD)
+
+#
+# SUIT encryption keys
+#
+
+# Specify key to use.
+# Will use $(SUIT_KEY_DIR)/$(SUIT_KEY) $(SUIT_KEY_DIR)/$(SUIT_KEY).pub as
+# private/public key files, similar to how ssh names its key files.
+SUIT_KEY ?= default
+
+ifeq (1, $(RIOT_CI_BUILD))
+  SUIT_KEY_DIR ?= $(BINDIR)
+else
+  SUIT_KEY_DIR ?= $(RIOTBASE)/keys
+endif
+
+SUIT_SEC ?= $(SUIT_KEY_DIR)/$(SUIT_KEY)
+SUIT_PUB ?= $(SUIT_KEY_DIR)/$(SUIT_KEY).pub
+
+SUIT_PUB_HDR = $(BINDIR)/riotbuild/public_key.h
+SUIT_PUB_HDR_DIR = $(dir $(SUIT_PUB_HDR))
+CFLAGS += -I$(SUIT_PUB_HDR_DIR)
+BUILDDEPS += $(SUIT_PUB_HDR)
+
+$(SUIT_SEC) $(SUIT_PUB): | $(CLEAN)
```

```suggestion
$(SUIT_SEC) $(SUIT_PUB): $(CLEAN)
```

If I understand `order-only prerequesites` (and I'm not sure I do), the current situation makes it that `$(SUIT_SEC) $(SUIT_PUB)` don't need to be recompiled even if `$(CLEAN)` has changed (in this case is executed). This would be no Issue since normally `$(SUIT_SEC) $(SUIT_PUB)`  will be recompiled since they would be deleted. But if `-j` is used make will see `$(SUIT_SEC) $(SUIT_PUB)` hase not changed and has no `normal pre-requisite` and since `$(CLEAN)` is order only it will execute them in parallel and every other compilation will fail.

At least compilation stopped failing for after changing that, I was running this command:

```
docker run -ti --rm -v $(pwd):/data/riotbuild -u $(id -u) riot/riotbuild:latest make RIOT_CI_BUILD=1 BOARD=samr21-xpro -Cexamples/suit_update clean all -j8
```

---

fixed

---

seem you forgot to commit ;)